libgnucash/engine/Account.cpp

```diff
@@ -4392,6 +4392,63 @@ xaccAccountSetReconcileLastDate (Account *acc, time64 last_date)
     xaccAccountCommitEdit (acc);
 }
 
+
+void
+xaccAccountClearReconcileStatementData (Account *acc)
+{
+    if (!acc) return;
+
+    qof_instance_slot_path_delete
+        (QOF_INSTANCE(acc), {KEY_RECONCILE_INFO, "statements-data"});
+}
+
+
+gboolean
+xaccAccountGetReconcileStatementBalanceAtDate (const Account *acc,
+                                               time64 statement_date,
+                                               gnc_numeric *ending_balance)
+{
+    std::string date_str = qof_print_date (statement_date);
+    gnc_numeric bal;
+    GValue v = G_VALUE_INIT;
+
+    g_return_val_if_fail(GNC_IS_ACCOUNT(acc), FALSE);
+
+    qof_instance_get_path_kvp (QOF_INSTANCE(acc), &v,
+                               {KEY_RECONCILE_INFO, "statements", date_str});
+
+    if (!G_VALUE_HOLDS_INT64 (&v))
+        return FALSE;
+
+    bal = *(gnc_numeric*) g_value_get_boxed (&v);
+
+    if (!bal.denom)
+        return FALSE;
+
+    if (ending_balance) *ending_balance = bal;
+    return TRUE;
+}
+
+void
+xaccAccountSetReconcileStatementBalanceAtDate (Account *acc,
+                                               time64 statement_date,
+                                               gnc_numeric ending_balance)
+{
+    std::string date_str = qof_print_date (statement_date);
```

Comment:
`qof_print_date` is probably not the best choice...

Reply:
`gnc_time64_to_iso8601_buff` is especially good because you can create the buffer on the stack, so no allocation.